pkg/egressgateway/manager_privileged_test.go

```diff
@@ -179,7 +174,7 @@ func (k *EgressGatewayTestSuite) TestEgressGatewayManager(c *C) {
 
 	// Create a new policy
 	policy1 := newEgressPolicyConfigWithNodeSelector("policy-1", ep1Labels, destCIDR, []string{}, nodeGroup1Selector, testInterface1)
-	egressGatewayManager.OnAddEgressPolicy(policy1)
+	egressGatewayManager.onAddEgressPolicy(policy1)
```

Comment:
I think we shouldn't call the method directly, but instead go via k.policies by sending an event. That means constructing `CEGP` in tests however. Currently there is only a helper for `PolicyConfig`

@jibi should I change this to something like the following?
```go
k.policies.process(resource.Event[*Policy]{Obj: CEGP{...}})
```